### DISCUSSION TITLE
is there a way to use desktop for compiling

### DISCUSSION THREAD

**Discussion by xiro-codes** [Jovian-Experiments/Jovian-NixOS#177] [Q&A] [answered]
building the kernel takes forever how can i off load this to my desktop ?

**Comment by nat522** [comment]
I don’t believe so

**Comment by samueldr** [comment, marked as answer]
Yes there is. There are a couple ways to achieve this. I won't go into details since I don't have the time, but here's some resources to start looking into it:

Using Nix distributed builds

 - In the Nix manual: https://nixos.org/manual/nix/stable/advanced-topics/distributed-builds.html
 - On the Wiki (user-contributed notes): https://nixos.wiki/wiki/Distributed_build

Or using `nixos-rebuild --build-host`

(From man nixos-rebuild)

```
--build-host host
        Instead of building the new configuration locally, use the specified host to perform the
        build. The host needs to be accessible with ssh, and must be able to perform Nix builds.
        If  the  option --target-host is not set, the build will be copied back to the local ma‐
        chine when done.

        Note that, if --no-build-nix is not specified, Nix will be built both  locally  and  re‐
        motely.  This  is because the configuration will always be evaluated locally even though
        the building might be performed remotely.

        You can include a remote user name in the host name (user@host). You can  also  set  ssh
        options by defining the NIX_SSHOPTS environment variable.
```

**Comment by K900** [comment]
I'm going to convert this to a discussion, as this isn't really an _issue_.

**Comment by xiro-codes** [comment]
cool ima try it out thanks

**Comment by xiro-codes** [comment]
this might be worth mentioning in the docs, thank again the build is so much faster now also im not worrying about temps

**Reply by samueldr** [reply]
Glad to hear it worked out for you (with only those quick bits of info).

* * *

(To make this more actionable for contributors)

It might be worth to have a "tips for newcomers to NixOS" page, outlining some of the issues people that are just starting hit, and direct them toward the appropriate resources.
